cmds/core/cmp/cmp.go

```diff
@@ -70,53 +70,50 @@ func openFile(name string) (*os.File, error) {
 }
 
 // cmp is defined to fail with exit code 2
-func failf(format string, a ...interface{}) {
-	fmt.Fprintf(os.Stderr, format, a...)
-	os.Exit(2)
+func failf(w io.Writer, format string, a ...interface{}) int {
+	fmt.Fprintf(w, format, a...)
+	return 2
 }
 
-func main() {
-	flag.Parse()
+func cmp(w io.Writer, args ...string) int {
 	var offset [2]int64
 	var f *os.File
 	var err error
 
-	fnames := flag.Args()
-
 	cmpUnits := unit.DefaultUnits
 
 	off, err := unit.NewUnit(cmpUnits)
 	if err != nil {
-		failf("Could not create unit based on mapping: %v\n", err)
+		failf(w, "Could not create unit based on mapping: %v\n", err)
```

Comment:
ah, wait, fail is no longer failing! 

I would remove the fail functions, it's pretty useless, and cmp should change to return (int, error), and just have every line with fail do a return count, error

The caller can exit with os.Exit(2) if needed.